prisma/schema.prisma

```diff
@@ -48,8 +87,12 @@ model User {
   ideas             Idea[]
   votes             Vote[]
   comments          Comment[]
+  collections       Collection[]
+  proposals         Proposal[]
 }
 
+// how do we want to keep track of edits?
```

Comment:
I guess we could have another table that is almost a duplicate of `Proposal` but associated via the proposal id.

```
model ProposalHistory {
  id          Int       @id @default(autoincrement())
  createdAt   DateTime  @default(now())
  title       String    @db.VarChar(50)
  description String    @db.VarChar(1080)
  // Any other editable fields like images etc...
  proposal    Proposal  @relation(fields: [proposalId], references: [id], onDelete: Cascade)
  proposalId  Int
}
```

Then each time a proposal is created + edited we just push into here so we have a way to go back through versions.

Alternatively we could just add a JSON field to the Proposal model and just stash the changes in there but that might be more difficult to implement rollbacks if we need to?